services/office-add-in-api/test/lambdas/fetchReferenceCode.test.js

```diff
@@ -0,0 +1,214 @@
+import { main } from '../../lambdas/fetchReferenceCode';
+import * as dynamoDb from '../../../../libs/dynamoDb';
+
+jest.mock('../../../../libs/dynamoDb');
+
+const mockHeaders = {
+  'Access-Control-Allow-Credentials': true,
+  'Access-Control-Allow-Origin': '*',
+};
+
+const mockContext = {
+  awsRequestId: 'xxxxx',
+};
+
+const mockUUID = '1a2b3c';
+const mockEmail = 'test@test.com';
+const mockPersonalNumber = '0101010101';
+
+const mockDyanmoGetResponse = {
+  Item: { uuid: mockUUID },
+};
+
+const mockDyanmoQueryResponse = {
+  Count: 1,
+  Items: [{ uuid: mockUUID }],
+};
+
+const mockDyanmoQueryNotFoundResponse = {
+  Count: 0,
+  Items: [],
+};
+
+const mockDyanmoGetNotFoundResponse = {
+  Item: undefined,
+};
+
+const mockResponseBody = {
+  jsonapi: {
+    version: '1.0',
+  },
+  data: {
+    type: 'userFetchReferenceCode',
+    referenceCode: mockUUID,
+  },
+};
+
+const mock404body = {
+  jsonapi: {
+    version: '1.0',
+  },
+  data: {
+    status: '404',
+    code: '404',
+    message: 'No user with that personal number or email found in the database.',
+  },
+};
+
+const mock500body = {
+  jsonapi: {
+    version: '1.0',
+  },
+  data: {
+    status: '500',
+    code: '500',
+    message: 'Internal server error',
+  },
+};
+
+const mock400body = {
+  jsonapi: {
+    version: '1.0',
+  },
+  data: {
+    status: '400',
+    code: '400',
+    message: 'Missing required parameter: "query"',
+  },
+};
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+it('gets reference code for email', async () => {
+  expect.assertions(2);
+
+  const mockEvent = {
+    pathParameters: {
+      query: mockEmail,
+    },
+  };
+
+  const expectedResult = {
+    body: JSON.stringify(mockResponseBody),
+    headers: mockHeaders,
+    statusCode: 200,
+  };
+
+  dynamoDb.call.mockResolvedValueOnce(mockDyanmoQueryResponse);
+
+  const result = await main(mockEvent, mockContext);
+
+  expect(result).toEqual(expectedResult);
+  expect(dynamoDb.call).toHaveBeenCalledTimes(1);
+});
+
+it('gets reference code for personal number', async () => {
+  expect.assertions(2);
+
+  const mockEvent = {
+    pathParameters: {
+      query: mockPersonalNumber,
+    },
+  };
+
+  const expectedResult = {
+    body: JSON.stringify(mockResponseBody),
+    headers: mockHeaders,
+    statusCode: 200,
+  };
+
+  dynamoDb.call.mockResolvedValueOnce(mockDyanmoGetResponse);
+
+  const result = await main(mockEvent, mockContext);
+
+  expect(result).toEqual(expectedResult);
+  expect(dynamoDb.call).toHaveBeenCalledTimes(1);
```

Comment:
I would like to see that dynamodb.call has been called with correct parameters. This because we do have logic that changes parameters depending if its email or not.

so:
```js
...
expect(dynamodb.call).toHaveBeenCalledWith({...});
```